src/test/groovy/com/mateusz/jakuszko/roomforyou/service/ApartmentDbServiceSpecification.groovy

```diff
@@ -0,0 +1,142 @@
+import com.mateusz.jakuszko.roomforyou.exceptions.NotFoundException
+import com.mateusz.jakuszko.roomforyou.repository.ApartmentRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import spock.lang.Specification
+
+@SpringBootTest
+class ApartmentDbServiceSpecification extends Specification {
+
+    @Autowired
+    private ApartmentDbService service
+    @Autowired
+    ApartmentRepository repository
+
+    def "should throw an exception when there isn't such apartment"() {
+        when: "pull apartment from database which doesn't exist"
+        Optional<Apartment> apartment = service.getApartment(123)
+        apartment.orElseThrow(NotFoundException::new)
+
+        then:
+        thrown(NotFoundException)
+    }
+
+    def "should save and load apartment from db"() {
+        given:
+        Apartment apartment = createApartment()
+
+        when:
+        def savedApartment = saveAndGetApartmentFromDb(apartment)
+
+        then:
+        notThrown(NotFoundException)
+        checkIfFieldsAreCorrect(savedApartment, apartment)
+    }
+
+    def "should save apartments and return list of apartments after pull from db"() {
+        given:
+        createAndSaveApartments()
+
+        when: "pull apartments from db"
+        List<Apartment> apartments = service.getApartments()
+
+        then: "size o apartments equals 2"
+        apartments.size() == 2
+        and: "assert all fields"
+        apartments.get(0).getId() != null
+        apartments.get(0).getLatitude() == 11
+        apartments.get(0).getLongitude() == 12
+        apartments.get(0).getStreet() == "WallStreet"
+        apartments.get(0).getStreetNumber() == "13"
+        apartments.get(0).getApartmentNumber() == 14
+    }
+
+    def "should update existing apartment"() {
+        given:
+        Apartment apartment = createApartment()
+        Apartment savedApartment = saveAndGetApartmentFromDb(apartment)
+
+        when:
+        Apartment updatedApartment = changeFieldsUpdateAndGetApartmentFromDb(savedApartment)
+
+        then:
+        notThrown(NotFoundException)
+        updatedApartment.getCity() == "Warsaw"
+        updatedApartment.getLatitude() == 23.2
+        updatedApartment.getLongitude() == 52.2
+        updatedApartment.getApartmentNumber() == 23
+        updatedApartment.getStreet() == "Pilsudskiego"
+        updatedApartment.getStreetNumber() == "43"
+    }
+
+    def "should not found an apartment when it hase been deleted"() {
+        given: "create and save apartment into database"
+        Apartment apartment = createApartment()
+        service.save(apartment)
+        Long id = apartment.getId()
+
+        when: "delete apartment and try to pull it from database"
+        service.delete(id)
+        service.getApartment(id).orElseThrow(NotFoundException::new)
+
+        then: "apartment shouldn't found"
+        thrown(NotFoundException)
```

Comment:
Tu podobnie jak wczesnie proponuję użyć Optionala bez rzucania wyjątku
```
        when: 
        service.delete(id)

        then:
        service.getApartment(id).isEmpty()
```
Dodatkowo zwróć uwagę że w "when" (zgodnie z tym co napisał gość w kapeluszu) chcemy mieć tylko akcję którą testujemy a pobieranie z bazy jest konieczne tylko po to żeby móc wykonać jakąś weryfikację więc powinno wylądować w "then"

Reply:
Poprawione. Będę teraz bardziej się nad tym skupiał